src/main/java/mate/academy/rickandmorty/service/impl/CharacterServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.rickandmorty.service.impl;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDto;
+import mate.academy.rickandmorty.dto.internal.CharacterDto;
+import mate.academy.rickandmorty.mapper.CharacterMapper;
+import mate.academy.rickandmorty.model.Character;
+import mate.academy.rickandmorty.repository.CharacterRepository;
+import mate.academy.rickandmorty.service.CharacterClient;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CharacterServiceImpl implements CharacterService {
+    private static Long maxId;
+    private static boolean isDbUpdated;
+    private final CharacterClient client;
+    private final CharacterMapper mapper;
+    private final CharacterRepository repository;
+
+    @Override
+    public CharacterDto getRandom() {
+        if (!isDbUpdated) {
+            getDataFromApi();
+        }
+        Random random = new Random();
+        long randomLong = random.nextLong(maxId + 1);
```

Comment:
```suggestion
        long randomLong = random.nextLong(maxId) + 1;
```
because nextLong() returns 0<=x<maxId